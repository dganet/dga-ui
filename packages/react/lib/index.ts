import { ButtonComponent } from "./components/Button/ButtonComponent";

export {
  ButtonComponent as Button,
}