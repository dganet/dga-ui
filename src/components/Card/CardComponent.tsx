import React from 'react'

import { useTheme } from '@hooks/useTheme'
import { Container } from './CardComponent.styles'
import { CardProps } from './CardComponent.types'

const CardComponent = ({ children, cardType }: CardProps) => {
    const theme = useTheme()
    return (
        <Container theme={theme} cardType={cardType}>
            {children}
        </Container>
    )
}

export default CardComponent
