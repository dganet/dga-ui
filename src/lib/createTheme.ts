import { defaultTheme } from '../styles/defaultTheme'

export const createTheme = (theme: typeof defaultTheme) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },

        name: theme.name || defaultTheme.name
    } as typeof defaultTheme
}
