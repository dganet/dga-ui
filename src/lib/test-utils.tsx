import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { defaultTheme } from '../styles/defaultTheme'
import { CustomThemeProvider } from '../hooks/useTheme/ThemeContext'

const Providers: FC = ({ children }) => {
    return (
        <CustomThemeProvider theme={defaultTheme}>
            {children}
        </CustomThemeProvider>
    )
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
