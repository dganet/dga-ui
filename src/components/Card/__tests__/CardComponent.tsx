import { render, screen } from '../../../lib/test-utils'

import CardComponent from '../CardComponent'

describe('<CardComponent />', () => {
  it('should render the component', () => {
    const { container } = render(<CardComponent />)

    expect(screen.getByRole('heading', { name: /Card/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
