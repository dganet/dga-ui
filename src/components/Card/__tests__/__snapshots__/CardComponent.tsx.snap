// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardComponent /> should render the component 1`] = `
<div
  class="sc-bdfBQB ccoSRg"
>
  test
</div>
`;
